File: mobile/library/kotlin/io/envoyproxy/envoymobile/filters/RequestFilterCallbacksImpl.kt
package io.envoyproxy.envoymobile

import io.envoyproxy.envoymobile.engine.types.EnvoyHTTPFilterCallbacks

/**
 * Envoy implementation of `RequestFilterCallbacks`.
 */
internal class RequestFilterCallbacksImpl constructor(
  internal val callbacks: EnvoyHTTPFilterCallbacks
) : RequestFilterCallbacks {

  override fun resumeRequest() {
    callbacks.resumeIteration()
  }

  override fun resetIdleTimer() {
    callbacks.resetIdleTimer()
  }
}

File: mobile/library/swift/filters/RequestFilterCallbacksImpl.swift
@_implementationOnly import EnvoyEngine
import Foundation

/// Envoy implementation of RequestFilterCallbacks
final class RequestFilterCallbacksImpl: NSObject {
  private let callbacks: EnvoyHTTPFilterCallbacks

  init(callbacks: EnvoyHTTPFilterCallbacks) {
    self.callbacks = callbacks
    super.init()
  }
}

extension RequestFilterCallbacksImpl: RequestFilterCallbacks {
  func resumeRequest() {
    self.callbacks.resumeIteration()
  }

  func resetIdleTimer() {
    self.callbacks.resetIdleTimer()
  }
}
